Fenrir/PackDeltaMethods.fs

```diff
@@ -0,0 +1,88 @@
+﻿module Fenrir.DeltaCommands
+
+open System.IO
+open Fenrir.Tools
+open Fenrir.Zlib
+
+let MSBCondition (b: byte): bool =
+    b >= 128uy
+
+let estimateSize (startSize: int) (off: int) (bytes: byte array): int =
+        let mutable shift = off - 7
+        Array.fold (fun acc elem ->
+            shift <- shift + 7
+            int (elem % 128uy) <<< shift ||| acc) startSize bytes
+
+let byteToBits (num: byte) : byte[] =
+    let mutable number = num
+    Array.init 8 (fun _ ->
+        let result = number % 2uy
+        number <- number >>> 1
+        result)
+
+
+let bitsToInt (bits: byte[]) =
+    let mutable st = 1
+
+    Array.foldBack (fun i acc ->
+        let result = acc + int i * st
+        st <- st * 2
+        result) bits 0
+
+let insertCommand (reader: BinaryReader) (mem: MemoryStream) (flag: byte) (counter: int) =
+    mem.Write(reader.ReadBytes(int flag), 0, int flag)
+    counter + int flag
+
+let copyCommand (comm: byte array) (mem: MemoryStream)
+    (delta: BinaryReader) (nonDelta: BinaryReader) (count: int) =
+
+    let mutable counter = count
+    let extractBit (flag: byte) =
+        match flag with
+            | 1uy ->
+                counter <- counter + 1
+                int <| delta.ReadByte()
+            | _ -> 0
+
+    let copyOffset =
+        extractBit comm.[7] +
+        (extractBit comm.[6] <<< 8) +
+        (extractBit comm.[5] <<< 16) +
+        (extractBit comm.[4] <<< 24)
+    let copySize =
+        extractBit comm.[3] +
+        (extractBit comm.[2] <<< 8) +
+        (extractBit comm.[1] <<< 16)
```

Comment:
In the documentation, there's also a special case of `copySize` being equals to `0x10000` if the read value was zero (since there's no point in making a zero-byte copy, I guess). I think we should take this special case into account here.

Reply:
Oof.
Slipped my mind.